include/mbedtls/mbedtls_config.h

```diff
@@ -3634,7 +3634,7 @@
 
 /* Platform options */
 //#define MBEDTLS_PLATFORM_STD_MEM_HDR   <stdlib.h> /**< Header to include if MBEDTLS_PLATFORM_NO_STD_FUNCTIONS is defined. Don't define if no header is needed. */
-//#define MBEDTLS_PLATFORM_STD_CALLOC        calloc /**< Default allocator to use, can be undefined */
+//#define MBEDTLS_PLATFORM_STD_CALLOC        calloc /**< Default allocator to use, can be undefined. Please note that it should zeroize the allocated buffer. */
```

Comment:
We tend to use “zeroize” to mean “wipe confidential data”. Here the emphasis is on setting the contents to 0 (setting random content would be wrong). So I'd prefer not to use the word “zeroize” in the documentation.

Also, if we're documenting this aspect of the function, we might as well document the rest:

* The size of the buffer is the product of the two parameters.
* The behavior is undefined if the product of the two parameters overflows `size_t`.
* If the product is 0, the function may either return `NULL` or a valid pointer to an array of size 0 which is a valid input to the deallocation function.
* The corresponding deallocation function is `MBEDTLS_PLATFORM_STD_FREE`.

And for free:

* `NULL` is a valid parameter, and the function must do nothing.
* A non-null parameter will always be a pointer previously returned by `MBEDTLS_PLATFORM_STD_CALLOC` and not yet freed.

Reply:
> We tend to use “zeroize” to mean “wipe confidential data”. Here the emphasis is on setting the contents to 0

"Zeroize" means "setting to 0". Yes, it's what we want to do with confidential data (and the issue there can be that compilers will remove what they perceive as a dead store), but there's nothing about "zeroize" that implies that the thing it's setting to zero is confidential.

(Actually overwriting confidential data with random values would be just as secure as setting zeroising, but would have higher cost.)

`calloc()` MUST set the memory to `0` otherwise things will fail, so it should be documented - you can't set `mbedtls_calloc()` to a function that returns `malloc(count * size)` (even ignoring the overflow problem).